theano/tensor/c_code/alt_blas_template.c

```diff
@@ -1,8 +1,43 @@
 /** Alternative template NumPy-based implementation of BLAS functions used in Theano. **/
 
+/* Compute tensor[i] = scalar for every position i in tensor. */
+void alt_numpy_memset_inplace_%(float_type)s(PyArrayObject* tensor, const %(float_type)s* scalar) {
+    if (PyArray_IS_C_CONTIGUOUS(tensor) && *scalar == (char)(*scalar)) {
+        // This will use memset.
+        PyArray_FILLWBYTE(tensor, (char)(*scalar));
+        return;
+    }
+    NpyIter* iterator = NpyIter_New(tensor,
```

Comment:
This seem a duplicate of alt_numpy_scale_matrix_inplace_%(float_type)s. Did you wanted to change line 24 too to do a affectation only?

Reply:
Yes, both functions have pretty similar code (because of NumPy iteration code). I have indeed changed line 24 because here I just want to do an affectation.

It is needed in some cases, e.g. when beta is 0. In such case output tensor may contain arbitrary values, so we must explicitly set them to 0, otherwise a nan might be multiplied by 0, resulting to still a nan in final output.